tests/rule_based_profiler/test_rule.py

```diff
@@ -192,3 +192,255 @@ def test_get_parameter_value_by_fully_qualified_parameter_name_valid_parameter_n
         )
         == 6.5e-1
     )
+
+    fully_qualified_parameter_name = "$parameter.monthly_taxi_fairs.mean_values"
+    assert (
+        get_parameter_value_by_fully_qualified_parameter_name(
+            fully_qualified_parameter_name=f"{fully_qualified_parameter_name}.value[0]",
+            domain=column_Date_domain,
+            variables=rule_with_variables_with_parameters.variables,
+            parameters=rule_with_variables_with_parameters.parameters,
+        )
+        == 2.3
+    )
+    assert get_parameter_value_by_fully_qualified_parameter_name(
+        fully_qualified_parameter_name=f"{fully_qualified_parameter_name}.details",
+        domain=column_Date_domain,
+        variables=rule_with_variables_with_parameters.variables,
+        parameters=rule_with_variables_with_parameters.parameters,
+    ) == {
+        "confidence": "low",
+    }
+
```

Comment:
Could we split up all these asserts into different tests? Or use `@pytest.mark.parametrize`? I appreciate the thoroughness here but the length of this test makes it hard to reason about.

Reply:
It looks like the only thing that is changing is the parameter_name and the return value? In that case I think the parameterize solution or a list of tuples are viable solutions here.

```
test_cases = [
    (f"{fully_qualified_parameter_name}.details", {"confidence": low"})
    ...
]

for fully_qualified_parameter_name, expected_result in test_cases:
    res = get_parameter_value_by_fully_qualified_parameter_name(
        fully_qualified_parameter_name=fully_qualified_parameter_name
        domain=column_Date_domain,
        variables=rule_with_variables_with_parameters.variables,
        parameters=rule_with_variables_with_parameters.parameters
    )
    assert res == expected_result
```

Somewhat of a similar solution to the test script runner's list of dicts?

Reply:
@cdkini I used `pytest.parametrize` and made the actual test method smaller.  Thanks.